code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -1312,5 +1319,150 @@
+/datum/reagent/medicine/metafactor/overdose_start(mob/living/carbon/M)
+	metabolization_rate = 2  * REAGENTS_METABOLISM
+
+/datum/reagent/medicine/metafactor/overdose_process(mob/living/carbon/M)
+	if(prob(25))
+		M.vomit()
+	..()
+
+/datum/reagent/medicine/rhigoxane
+	name = "Rhigoxane"
+	description = "A second generation burn treatment agent exibiting a cooling effect that is especially pronounced when deployed as a spray. It's high halogen content helps extiguish fires."
+	reagent_state = LIQUID
+	color = "#B6D2F2"
+	overdose_threshold = 25
+	reagent_weight = 0.6
+
+/datum/reagent/medicine/rhigoxane/on_mob_life(mob/living/carbon/M)
+	M.adjustFireLoss(-2*REM, 0.)
+	M.adjust_bodytemperature(-20 * TEMPERATURE_DAMAGE_COEFFICIENT, BODYTEMP_NORMAL)
+	..()
+	. = 1
+
+/datum/reagent/medicine/rhigoxane/reaction_mob(mob/living/carbon/M, method=VAPOR, reac_volume)
+	if(method != VAPOR)
+		return
+
+	M.adjust_bodytemperature(-reac_volume * TEMPERATURE_DAMAGE_COEFFICIENT * 20, 200)
+	M.adjust_fire_stacks(-reac_volume / 2)
+	if(reac_volume >= metabolization_rate)
+		M.ExtinguishMob()
+
+	..()
+
+/datum/reagent/medicine/rhigoxane/overdose_process(mob/living/carbon/M)
+	M.adjustFireLoss(3*REM, 0.)
+	M.adjust_bodytemperature(-35 * TEMPERATURE_DAMAGE_COEFFICIENT, 50)
+	..()
+
+/datum/reagent/medicine/thializid
+	name = "Thializid"
+	description = "A potent antidote for intravenous use with a narrow therapeutic index, it is considered an active prodrug of oxalizid."
+	reagent_state = LIQUID
+	color = "#8CDF24" // heavy saturation to make the color blend better
+	metabolization_rate = 0.75 * REAGENTS_METABOLISM
+	overdose_threshold = 6
+	var/conversion_amount
+
+/datum/reagent/medicine/thializid/reaction_mob(mob/living/carbon/M, method=INJECT, reac_volume)
+	if(method != INJECT)
+		return
+	if(reac_volume >= 0.6) //prevents cheesing with ultralow doses.
+		M.adjustToxLoss(-1.5 * min(2, reac_volume) * REM, 0)	  //This is to promote iv pole use for that chemotherapy feel.
+	if(!locate(/obj/item/organ/liver) in M.internal_organs)
+		return
+	var/obj/item/organ/liver/L = M.internal_organs_slot[ORGAN_SLOT_LIVER]
+	if(L.failing)
+		return
+	conversion_amount = reac_volume	* (min(100 -L.damage, 80) / 100) //the more damaged the liver the worse we metabolize
+	to_chat(M, "conversion amount:[conversion_amount]")
+	to_chat(M, "reaction volume:[reac_volume]")
+	to_chat(M, "Liver damage:[L.damage]")
+	M.reagents.remove_reagent(/datum/reagent/medicine/thializid, conversion_amount)
+	M.reagents.add_reagent(/datum/reagent/medicine/oxalizid, conversion_amount)
+	..()
+
+/datum/reagent/medicine/thializid/on_mob_life(mob/living/carbon/M)
+	M.adjustToxLoss(-1*REM, 0)
+	for(var/datum/reagent/toxin/R in M.reagents.reagent_list)
+		M.reagents.remove_reagent(R.type,1)
+
+	M.reagents.add_reagent("oxalizid", 0.075 * REM) //add a little bit of the metabolite after metabolism
+	..()
+	. = 1
+
+/datum/reagent/medicine/thializid/overdose_process(mob/living/carbon/M)
+	M.adjustLiverLoss(2)
```

Comment:
You can do liver damage by doing M/human.applyLiverDamage(25), though I'm not sure how kosher that is with the coders. Just kinda find it weird to add duplicates of liver procs. Can use this to get liver damage, too.